GitUI/CommandsDialogs/UserScriptContextMenuExtensions.cs

```diff
@@ -10,90 +10,100 @@ namespace GitUI.CommandsDialogs
-                        Text = script.Name,
-                        Name = script.Name + "_ownScript",
-                        Image = script.GetIcon(),
-                        ShortcutKeyDisplayString = Hotkeys.Value?.FirstOrDefault(h => h.Name == script.Name)?.KeyData.ToShortcutKeyDisplayString()
-                    };
-
-                    item.Click += (s, e) =>
-                    {
-                        string scriptKey = script.Name;
-                        scriptInvoker(scriptKey);
-                    };
-
-                    if (script.AddToRevisionGridContextMenu)
-                    {
-                        contextMenu.Items.Add(item);
-                    }
-                    else
-                    {
-                        hostMenuItem.DropDown.Items.Add(item);
-                    }
-                }
+            bool showScriptsMenu = hostMenuItem.DropDown.Items.Count > 0;
+            hostMenuItem.Visible = showScriptsMenu;
+        }
 
-                if (lastIndex != contextMenu.Items.Count)
-                {
-                    contextMenu.Items.Insert(lastIndex, new ToolStripSeparator());
-                }
+        /// <summary>
+        ///  Removes user scripts from the <paramref name="contextMenu"/>, or under <paramref name="hostMenuItem"/>,
+        ///  if scripts are marked as <see cref="ScriptInfo.AddToRevisionGridContextMenu"/>.
+        /// </summary>
+        /// <param name="contextMenu">The context menu to add user scripts too.</param>
+        /// <param name="hostMenuItem">The menu item to which to add user scripts marked as <see cref="ScriptInfo.AddToRevisionGridContextMenu"/>.</param>
+        public static void RemoveUserScripts(this ContextMenuStrip contextMenu, ToolStripMenuItem hostMenuItem)
+        {
+            contextMenu = contextMenu ?? throw new ArgumentNullException(nameof(contextMenu));
+            hostMenuItem = hostMenuItem ?? throw new ArgumentNullException(nameof(hostMenuItem));
+
+            RemoveOwnScripts(contextMenu, hostMenuItem);
+        }
 
-                bool showScriptsMenu = hostMenuItem.DropDown.Items.Count > 0;
-                hostMenuItem.Visible = showScriptsMenu;
+        private static void RemoveOwnScripts(ContextMenuStrip contextMenu, ToolStripMenuItem hostMenuItem)
+        {
+            hostMenuItem.DropDown.Items.Clear();
+
+            var list = contextMenu.Items.Cast<ToolStripItem>()
+                .Where(x => x.Name.EndsWith(ScriptNameSuffix))
+                .ToList();
+
+            foreach (var item in list)
+            {
+                contextMenu.Items.RemoveByKey(item.Name);
             }
+
+            if (contextMenu.Items[contextMenu.Items.Count - 1] is ToolStripSeparator)
+            {
+                contextMenu.Items.RemoveAt(contextMenu.Items.Count - 1);
+            }
+
+            bool showScriptsMenu = hostMenuItem.DropDown.Items.Count > 0;
+            hostMenuItem.Visible = showScriptsMenu;
```

Comment:
It works without `.ToList()`:
```suggestion
            hostMenuItem.DropDown.Items.Clear();
            hostMenuItem.Visible = false;

            contextMenu.Items.Cast<ToolStripItem>()
                .Where(item => item.Name.EndsWith(ScriptNameSuffix))
                .ForEach(item => contextMenu.Items.RemoveByKey(item.Name));

            if (contextMenu.Items[contextMenu.Items.Count - 1] is ToolStripSeparator)
            {
                contextMenu.Items.RemoveAt(contextMenu.Items.Count - 1);
            }
```

Reply:
![image](https://user-images.githubusercontent.com/3169265/96114300-1c049600-0eee-11eb-89d7-01232cde0c02.png)

![image](https://user-images.githubusercontent.com/3169265/96114249-0becb680-0eee-11eb-96d4-58e18dc20396.png)

Reply:
Thank you for the explanation!

Reply:
So one final nit from my side:
```suggestion
            hostMenuItem.DropDown.Items.Clear();
            hostMenuItem.Visible = false;

            var list = contextMenu.Items.Cast<ToolStripItem>()
                .Where(x => x.Name.EndsWith(ScriptNameSuffix))
                .ToList();

            foreach (var item in list)
            {
                contextMenu.Items.RemoveByKey(item.Name);
            }

            if (contextMenu.Items[contextMenu.Items.Count - 1] is ToolStripSeparator)
            {
                contextMenu.Items.RemoveAt(contextMenu.Items.Count - 1);
            }
```